modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -242,12 +255,35 @@ inline void preprocess(const cv::Mat& src,
             dst = pp;
         }
 
+        if (with_pad) {
+            const auto div = ti.pad->divisibility;
+            const int padded_h = std::ceil(new_h / div) * div;
+            const int padded_w = std::ceil(new_w / div) * div;
+            int type = -1;
+            // is_hwc - Mat has C*H, W dimensions
+            if (!is_hwc) {
+                type = ddepth;
+            } else {
+                type = CV_MAKETYPE(ddepth, new_c);
+            }
+            GAPI_Assert(type != -1);
+            cv::Mat pad_im(cv::Size(padded_w, (!is_hwc ? new_c : 1) * padded_h), type, 0.f);
```

Comment:
The padding value (where you hardcode it to `0.f`) may need to be customizable.

Also -- the padding policy may need to be customizable. What you have here is an analogue to `BORDER_CONST`. There're also `MIRROR`, `REPLICATE`, and so on are defined for image processing.

Reply:
This may be a separate task, if the current API is enough to complete the current work/sample.

Reply:
Removed after "padding" discussion.